operations/security_and_authentication/Add_LDAP_User_Federation.md

```diff
@@ -1,32 +1,71 @@
 
+
 ### Prerequisites
 
 LDAP user federation is not currently configured in Keycloak. For example, if it was not configured in Keycloak when the system was initially installed or the LDAP user federation was removed.
 
+
 ### Procedure
 
-1.  Update the LDAP settings in the customizations.yaml file.
+1. Prepare to customize the customizations.yaml file.
 
-    The LDAP server CA certificate goes into SealedSecret. Refer to the "Generate Sealed Secrets" header in the [Prepare Site Init](../../install/prepare_site_init.md) procedure for instructions on how to do that.
+   If the customizations.yaml file is managed in an external Git repository (as recommended), then clone a local working tree. Replace the `<URL>` value in the following command before running it.
 
-    1.  Update the customizations.yaml file.
+   ```bash
+   ncn-m001# git clone <URL> /root/site-init
+   ncn-m001# cd /root/site-init
+   ```
 
-        LDAP connection information is stored in the keycloak-users-localize Secret in the services namespace. In the customizations.yaml file, set the values for the keycloak\_users\_localize keys in the spec.kubernetes.sealed\_secrets field:
+   If there is not a backup of site-init, perform the following steps to create a new one using the values stored in the Kubernetes cluster.
 
-        -   The ldap\_connection\_url key is required and is set to an LDAP URL.
-        -   The ldap\_bind\_dn and ldap\_bind\_credentials keys are optional.
-        -   If the LDAP server allows anonymous searches of users and groups, then these keys must not be set.
-        -   If the LDAP server requires authentication. then the bind DN and credentials are set in these keys respectively.
-        For example:
+   1. Create a new site-init directory using from the CSM tarball.
 
-        ```bash
-        keycloak_users_localize:
+      ```bash
+      ncn-m001# cp -r ${CSM_DISTDIR}/shasta-cfg /root/site-init
+      ncn-m001# cd /root/site-init
+      ```
+  
+   1. Extract customizations.yaml from the site-init secret.
+
+      ```bash
+      ncn-m001# kubectl -n loftsman get secret site-init -o jsonpath='{.data.customizations\.yaml}' | base64 -d - > customizations.yaml
+      ```
+
+   1. Extract the certificate and key used to create the sealed secrets.
+
+      ```bash
+      ncn-m001# mkdir certs
+      ncn-m001# kubectl -n kube-system get secret sealed-secrets-key -o jsonpath='{.data.tls\.crt}' | base64 -d - > certs/sealed_secrets.crt
+      ncn-m001# kubectl -n kube-system get secret sealed-secrets-key -o jsonpath='{.data.tls\.key}' | base64 -d - > certs/sealed_secrets.key
+      ```
+
+  > **NOTE:** All subsequent steps of this procedure should be performed within the `/root/site-init` directory created in this step.
+   
+2. Update the LDAP settings in the customizations.yaml file.
+
+   1. Modify the customizations.yaml file to put the LDAP server CA certificate into the SealedSecret.
```

Comment:
The procedure to modify the customizations.yaml file to put the LDAP server CA certificate into the SealedSecret. is described in install/prepare_site_init.md -- it's in step 8 "To federate Keycloak with an upstream LDAP:", and goes from "In the example below, the LDAP server has the hostname dcldap2.us.cray.com and is using the port 636." to "Then, inject and encrypt certs.jks.b64 into customizations.yaml:"